mars/dataframe/core.py

```diff
@@ -1837,6 +1837,78 @@ def itertuples(self, index=True, name='Pandas', batch_size=1000, session=None):
         """
         return self._data.itertuples(batch_size=batch_size, session=session,
                                      index=index, name=name)
+    
+    def assign(self, **kwargs):
+        """
+        Assign new columns to a DataFrame.
+        Returns a new object with all original columns in addition to new ones.
+        Existing columns that are re-assigned will be overwritten.
+        
+        Parameters
+        ----------
+        **kwargs : dict of {str: callable or Series}
+            The column names are keywords. If the values are
+            callable, they are computed on the DataFrame and
+            assigned to the new columns. The callable must not
+            change input DataFrame (though pandas doesn't check it).
+            If the values are not callable, (e.g. a Series, scalar, or array),
+            they are simply assigned.
+        
+        Returns
+        -------
+        DataFrame
+            A new DataFrame with the new columns in addition to
+            all the existing columns.
+        
+        Notes
+        -----
+        Assigning multiple columns within the same ``assign`` is possible.
+        Later items in '\*\*kwargs' may refer to newly created or modified
+        columns in 'df'; items are computed and assigned into 'df' in order.
+        
+        Examples
+        --------
+        >>> import mars.dataframe as md
+        >>> md = md.DataFrame({'temp_c': [17.0, 25.0]},
+        ...                   index=['Portland', 'Berkeley'])
+        >>> md.execute()
+                  temp_c
+        Portland    17.0
+        Berkeley    25.0
+        
+        Where the value is a callable, evaluated on `md`:
+        
+        >>> md.assign(temp_f=lambda x: x.temp_c * 9 / 5 + 32)
+        >>> md.execute()
+                  temp_c  temp_f
+        Portland    17.0    62.6
+        Berkeley    25.0    77.0
+        
+        Alternatively, the same behavior can be achieved by directly
+        referencing an existing Series or sequence:
+        
+        >>> md.assign(temp_f=df['temp_c'] * 9 / 5 + 32)
+        >>> md.execute()
+                  temp_c  temp_f
+        Portland    17.0    62.6
+        Berkeley    25.0    77.0
+        
+        You can create multiple columns within the same assign where one
+        of the columns depends on another one defined within the same assign:
+        
+        >>> md.assign(temp_f=lambda x: x['temp_c'] * 9 / 5 + 32,
+        ...           temp_k=lambda x: (x['temp_f'] +  459.67) * 5 / 9)
+        >>> md.execute()
+                  temp_c  temp_f  temp_k
+        Portland    17.0    62.6  290.15
+        Berkeley    25.0    77.0  298.15
+        """
+        
+        data = self.copy()
+
+        for k, v in kwargs.items():
+            data[k] = pd.core.common.apply_if_callable(v, data)
```

Comment:
We do not recommend using core module in pandas as it is not stable across different versions. Try using your own code to implement it.

Reply:
sure. Let me try